darwin/importer/formats/nifti.py

```diff
@@ -0,0 +1,288 @@
+            nifti_annotation["image"],
+            path,
+            class_map=nifti_annotation.get("class_map"),
+            mode=nifti_annotation.get("mode", "image"),
+        )
+        annotation_files.append(annotation_file)
+    return annotation_files
+
+
+def _parse_nifti(nifti_path: Path, filename: Path, json_path: Path, class_map: Dict, mode: str) -> dt.AnnotationFile:
+
+    img: np.ndarray = process_nifti(nib.load(nifti_path))
+
+    shape = img.shape
+    processed_class_map = process_class_map(class_map)
+    video_annotations = []
+    if mode == "instances":  # For each instance produce a video annotation
+        for class_name, class_idxs in processed_class_map.items():
+            if class_name == "background":
+                continue
+            class_img = np.isin(img, class_idxs).astype(np.uint8)
+            cc_img, num_labels = cc3d.connected_components(class_img, return_N=True)
+            for instance_id in range(1, num_labels):
+                video_annotation = get_video_annotation(cc_img, class_idxs=[instance_id], class_name=class_name)
+                if video_annotation:
+                    video_annotations.append(video_annotation)
+    elif mode == "image":  # For each frame and each class produce a single frame video annotation
+        for i in range(shape[-1]):
+            slice_mask = img[:, :, i].astype(np.uint8)
+            for class_name, class_idxs in processed_class_map.items():
+                frame_annotations = {}
+                if class_name == "background":
+                    continue
+                class_mask = np.isin(slice_mask, class_idxs).astype(np.uint8).copy()
+                polygon = mask_to_polygon(mask=class_mask, class_name=class_name)
+                if polygon is None:
+                    continue
+                frame_annotations[i] = polygon
+                video_annotation = dt.make_video_annotation(
+                    frame_annotations,
+                    keyframes={i: True, i + 1: True},
+                    segments=[[i, i + 1]],
+                    interpolated=False,
+                    slot_names=[],
+                )
+                video_annotations.append(video_annotation)
+    elif mode == "video":  # For each class produce a single video annotation
+        for class_name, class_idxs in processed_class_map.items():
+            if class_name == "background":
+                continue
+            video_annotation = get_video_annotation(img, class_idxs=class_idxs, class_name=class_name)
+            if video_annotation is None:
+                continue
+            video_annotations.append(video_annotation)
+    annotation_classes = set(
+        [dt.AnnotationClass(class_name, "polygon", "polygon") for class_name in class_map.values()]
+    )
+    return dt.AnnotationFile(
+        path=json_path,
+        filename=str(filename),
+        annotation_classes=annotation_classes,
+        annotations=video_annotations,
```

Comment:
`slots` should also be added, as fields like `filename` should be deprecated very soon. All V2 logic uses or should use data from `dt.Slot` not from root of `AnnotationFile`. This code doesn't crash only because right now use-case using it (import) doesn't need this metadata, but I think this is not a reason to create broken `AnnotationFile`. I know `slot_name` is not known here, and for that we still create `Slot` with `name: None`.

Reply:
You can take a look at how Darwin JSON 1.0 is read and decoded and do the same.

Reply:
Hey Rafal! I'm not sure how to add the slots here. It works without any of this metadata, and the `slot_name` is not known here. Wouldn't know what to fill in for any of these fields (except maybe width and height, but not sure how that feeds in to any frontend process):

```
dt.Slot(
            name=data["slot_name"],
            type=data["type"],
            width=data.get("width"),
            height=data.get("height"),
            source_files=data.get("source_files", []),
            thumbnail_url=data.get("thumbnail_url"),
            frame_count=data.get("frame_count"),
            frame_urls=data.get("frame_urls"),
            fps=data.get("fps"),
        ),
  ```
  Should I define an empty Slot object and pass it?